src/proof/dot/dot_printer.cpp

```diff
@@ -209,8 +230,45 @@ uint64_t DotPrinter::printInternal(std::ostream& out,
 
     pfLet[currentHash] = currentRuleID;
   }
+
+  ProofNodeClusterType proofNodeType = ProofNodeClusterType::NOT_DEFINED;
+  if (options::printDotClusters())
+  {
+    // Define the type of this node
+    proofNodeType = defineProofNodeType(pn, parentType);
+    if (proofNodeType != ProofNodeClusterType::FIRST_SCOPE
+        && proofNodeType != ProofNodeClusterType::NOT_DEFINED)
+    {
+      d_subgraphsStr[(int)proofNodeType - 1] << d_ruleID << " ";
+    }
+  }
+
   d_ruleID++;
 
+  printProofNodeInfo(out, pn, currentRuleID);
```

Comment:
move this to before the line above and you can use `d_ruleID` directly in `printProofNodeInfo`